src/tpm2_wrap.c

```diff
@@ -6087,5 +6125,561 @@ int wolfTPM2_ClearCryptoDevCb(WOLFTPM2_DEV* dev, int devId)
+        }
+        else {
+            XMEMCPY(policyDigest, policyGetDigestOut->policyDigest.buffer,
+                policyGetDigestOut->policyDigest.size);
+        }
+
+        *policyDigestSz = policyGetDigestOut->policyDigest.size;
+    }
+
+    return rc;
+}
+
+int wolfTPM2_PolicyPCR(TPM_HANDLE sessionHandle, TPM_ALG_ID pcrAlg,
+    word32* pcrArray, word32 pcrArraySz)
+{
+    int rc;
+    PolicyPCR_In policyPcr[1];
+
+    if (pcrArray == NULL || pcrArraySz == 0)
+        return BAD_FUNC_ARG;
+
+    XMEMSET(policyPcr, 0, sizeof(PolicyPCR_In));
+
+    /* add PolicyPCR to the policy */
+    policyPcr->policySession = sessionHandle;
+    TPM2_SetupPCRSelArray(&policyPcr->pcrs, pcrAlg, pcrArray,
+        pcrArraySz);
+
+    rc = TPM2_PolicyPCR(policyPcr);
+
+    return rc;
+}
+
+/* use public authKey, must have digest, signature and nonce from external */
+int wolfTPM2_SealWithAuthSig(WOLFTPM2_DEV* dev, WOLFTPM2_KEYBLOB* authKey,
+    WOLFTPM2_HANDLE* parent, TPMT_PUBLIC* template, WOLFTPM2_KEYBLOB* sealBlob,
+    TPM_HANDLE sessionHandle, TPM_ALG_ID pcrAlg, word32* pcrArray,
+    word32 pcrArraySz, const byte* sealData, word32 sealSz, byte* policyDigest,
+    word32 policyDigestSz, const byte* nonce, word32 nonceSz,
+    const byte* policyDigestSig, word32 policyDigestSigSz)
+{
+    int rc;
+    byte checkDigest[TPM_MAX_DIGEST_SIZE];
+    word32 checkDigestSz = TPM_MAX_DIGEST_SIZE;
+    PolicyAuthorize_In policyAuthIn[1];
+    WOLFTPM2_HASH hash[1];
+    TPM_ALG_ID sigAlg = TPM_ALG_NULL;
+
+    if (dev == NULL || authKey == NULL || parent == NULL || template == NULL ||
+        sealBlob == NULL || pcrArray == NULL ||
+        sealData == NULL || policyDigest == NULL || policyDigestSig == NULL) {
+        return BAD_FUNC_ARG;
+    }
+
+    XMEMSET(policyAuthIn, 0, sizeof(PolicyAuthorize_In));
+    XMEMSET(hash, 0, sizeof(WOLFTPM2_HASH));
+
+    /* PolicyPCR should already have been added to the policy */
+    /* start hash */
+    rc = wolfTPM2_HashStart(dev, hash, pcrAlg, NULL, 0);
+
+    /* hash the policyDigest */
+    if (rc == 0)
+        rc = wolfTPM2_HashUpdate(dev, hash, policyDigest, policyDigestSz);
+
+    /* hash the nonce */
+    if (rc == 0 && nonce != NULL)
+        rc = wolfTPM2_HashUpdate(dev, hash, nonce, nonceSz);
+
+    /* get the digest */
+    if (rc == 0)
+        rc = wolfTPM2_HashFinish(dev, hash, checkDigest, &checkDigestSz);
+
+    /* verify the signature, get the auth ticket, save it to policyAuthIn */
+    if (rc == 0) {
+
+        if (authKey->pub.publicArea.type == TPM_ALG_ECC) {
+            sigAlg = authKey->pub.publicArea.parameters.eccDetail.scheme.scheme;
+
+            if (sigAlg == TPM_ALG_NULL)
+                sigAlg = TPM_ALG_ECDSA;
+        }
+        else if (authKey->pub.publicArea.type == TPM_ALG_RSA) {
+            sigAlg = authKey->pub.publicArea.parameters.rsaDetail.scheme.scheme;
+
+            if (sigAlg == TPM_ALG_NULL)
+                sigAlg = TPM_ALG_RSASSA;
+        }
```

Comment:
What about `else` case? Should it error?

Reply:
I just want to pick a default algorithm if scheme is TPM_ALG_NULL, if the scheme is non null the previous lines will set sigAlg to a valid value